apps/climatemappedafrica/src/lib/hurumap/index.js

```diff
@@ -160,3 +163,84 @@ export async function fetchProfileGeography(
     items: formatProfileGeographyData(data, parent),
   };
 }
+
+export async function fetchCachedProfile(params, maxCacheAge = 10) {
+  const { profileId } = params;
+  const cacheDir = path.resolve(process.cwd(), "public", "cache");
+  const cacheFile = path.join(cacheDir, `profile-${profileId}.json`);
+
+  try {
+    await fs.mkdir(cacheDir, { recursive: true });
+
+    try {
+      const cachedData = await fs.readFile(cacheFile, "utf-8");
+      const { data, timestamp } = JSON.parse(cachedData);
+
+      const cacheAge = Date.now() - timestamp;
+      const maxCacheAgeInMs = maxCacheAge * 60 * 1000;
+
+      if (cacheAge < maxCacheAgeInMs) {
+        return data;
+      }
+    } catch (error) {
+      console.error(`No valid cache found for profile ${profileId}`);
+    }
+
+    const profileData = await fetchProfile(params);
+
+    const cacheContent = JSON.stringify({
+      timestamp: Date.now(),
+      data: profileData,
+    });
+    await fs.writeFile(cacheFile, cacheContent, "utf-8");
+
+    return profileData;
+  } catch (error) {
+    return fetchProfile(params);
+  }
+}
+
+export async function fetchCachedProfileGeography(
+  geoCode,
+  params,
+  maxCacheAge = 10,
+) {
+  const { profileId, version = "Climate" } = params;
+  const cacheDir = path.resolve(process.cwd(), "public", "cache");
+  const cacheFile = path.join(
+    cacheDir,
+    `geography-${profileId}-${geoCode.toUpperCase()}-${version}.json`,
+  );
+
+  try {
+    await fs.mkdir(cacheDir, { recursive: true });
+
+    try {
```

Comment:
I think separating this to a different function would be ideal @kelvinkipruto nesting try catch blocks is a symptom of code smells .